alchimia/engine.py

```diff
@@ -1,26 +1,58 @@
 from __future__ import absolute_import, division
 
 from sqlalchemy.engine.base import Engine
+from sqlalchemy.exc import StatementError
 
-from twisted.internet.threads import deferToThreadPool
+from twisted._threads import ThreadWorker, AlreadyQuit
+
+from twisted.internet.defer import Deferred, fail
+from twisted.python.failure import Failure
+
+from threading import Thread
+
+try:
+    from Queue import Queue
+except:
+    from queue import Queue
+
+
+def _threaded_worker():
+    def _start_thread(target):
+        thread = Thread(target=target)
+        thread.daemon = True
+        return thread.start()
+    return ThreadWorker(_start_thread, Queue())
+
+
+def _defer_to_worker(deliver, worker, work, *args, **kwargs):
+    deferred = Deferred()
+
+    @worker.do
+    def container():
+        try:
+            result = work(*args, **kwargs)
+        except:
+            f = Failure()
+            deliver(lambda: deferred.errback(f))
+        else:
+            deliver(lambda: deferred.callback(result))
+    return deferred
```

Comment:
Arguably there should be something like `deferToWorker` in `twisted._threads`, but discovering this kind of shortcoming is exactly what the "provisional" (private) status of this API is for :)